subsystem/kubeapi_test.go

```diff
@@ -264,4 +362,31 @@ var _ = Describe("[kube-api]cluster installation", func() {
 			return conditionsv1.IsStatusConditionTrue(getAgentCRD(ctx, kubeClient, key).Status.Conditions, v1alpha1.AgentSyncedCondition)
 		}, "2m", "10s").Should(Equal(true))
 	})
+
+	It("WIP: deploy clusterDeployment and installEnv and verify updates", func() {
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		clusterDeploymentSpec := getDefaultClusterDeploymentSNOSpec(secretRef)
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForClusterReconcileTimeout)
```

Comment:
suggesting to switch between this line and the next, if you expect to a state then the cluster will exist, here you don't really know when will the reconcile will be called and if cluster is already created.

Reply:
Done.
To make `waitForClusterDeploymentCRDState` not to throw nil pointer exception initially, I've added a check for the length of `clusterDeployment.Status.Conditions`